### PR TITLE
Add mill-mima to check binary compatibility

### PR DESCRIPTION
**Pull request by lolgab**
I developed this plugin so we can check our ecosystem and avoid breakages of binary compatibility.
I think that at the point of maturity this ecosystem is, we should start to check for binary compatibility and avoid breaking it.
Let me know what you think :)

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -18,9 +20,16 @@ val scalaNativeVersions = for {
...
-trait UtestModule extends PublishModule {
+trait UtestModule extends PublishModule with Mima {
   def artifactName = "utest"
 
+  def mimaPreviousArtifacts = Agg.from(
+    VcsVersion
+      .vcsState()
+      .lastTag
+      .map(lastTag => ivy"com.lihaoyi::utest::$lastTag")
+  )
```
Comment (lolgab) [inline build.sc:31]: This makes `mimaReportBinaryIssues` check for binary compatibility with the last tag on Git.

**Review by lefou — APPROVED** [review]
Looks good to me.

As a first step. I think later, maybe after the next mayor release, we want to check against all versions, esp. the first of a mayor release.

**Comment by lolgab** [comment]
@lefou Isn't it the point of major releases to be breaking and without binary compatibility guarantees? Or do you mean that, for example, `1.3.1` should check for both `1.3.0` and `1.0.0`?

**Comment by lefou** [comment]
I meant, that we may need to configure more than one version, not only the last tag. As you already suggested, for `1.3.3` we should addionally check against `1.3.0` and probably also `1.0.0`.

**Comment by lefou** [comment]
Warning: currently used mill version in this repo is 0.9.6-xx but mill-mima requires mill 0.9.8!

**Comment by lolgab** [comment]
Another nice thing to check would be forward compatibility for patch releases, but for that the plugin should be smarter, I guess.

**Review by anatoliykmetyuk — APPROVED** [review]

**Comment by lolgab** [comment]
I'm experiencing a strange error. Locally the `__.mimaReportBinaryIssues` completes fine. But on Github Actions it fails to get the last tag, failing with:
```
[42/714] de.tobiasroeser.mill.vcs.version.VcsVersion.vcsState 
fatal: no tag exactly matches 'ffb80f4d367a93f42bb19122894c877b0288bf4c'
fatal: No names found, cannot describe anything.
```
@lefou Have you ever experienced this?

**Comment by lefou** [comment]
> I'm experiencing a strange error. Locally the `__.mimaReportBinaryIssues` completes fine. But on Github Actions it fails to get the last tag, failing with:
> 
> ```
> [42/714] de.tobiasroeser.mill.vcs.version.VcsVersion.vcsState 
> fatal: no tag exactly matches 'ffb80f4d367a93f42bb19122894c877b0288bf4c'
> fatal: No names found, cannot describe anything.
> ```
> 
> @lefou Have you ever experienced this?

This is probably because GH by default does not check out the full history but only some (50?) commits. If your tag is more commits in the past, you won't see any tags. To change, add the `fetch-depth` option to the checkout action:

```yaml
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

**Comment by lolgab** [comment]
@lefou Gotcha! Thank you!

**Comment by lolgab** [comment]
Thank you for the reviews!
Now I'm going to add the plugin to the various com-lihaoyi repositories!

**Pull request merged**